pmd-core/src/test/java/net/sourceforge/pmd/lang/ast/xpath/AttributeAxisIteratorTest.java

```diff
@@ -4,19 +4,30 @@
 
 package net.sourceforge.pmd.lang.ast.xpath;
 
+import static junit.framework.TestCase.assertTrue;
+
 import java.util.HashMap;
 import java.util.Map;
 
 import org.junit.Assert;
 import org.junit.Test;
 
 import net.sourceforge.pmd.lang.ast.DummyNode;
+import net.sourceforge.pmd.lang.ast.DummyNodeWithDeprecatedAttribute;
+import net.sourceforge.pmd.lang.ast.Node;
+
 
 /**
  * Unit test for {@link AttributeAxisIterator}
  */
 public class AttributeAxisIteratorTest {
 
+    @Test
+    public void testAttributeDeprecation() {
+        Node dummy = new DummyNodeWithDeprecatedAttribute(2);
+        assertTrue(toMap(new AttributeAxisIterator(dummy)).containsKey("Size"));
```

Comment:
possible using `assertThat(toMap(new AttributeAxisIterator(dummy)), IsMapContaining.hasKey("Size"));` to have a more explicit test / better error messages. You may need to upgrade (add? I know we have it but may be transitionally through Mockito) the Hamcrest test dependency to be able to use this.